src/BuildIt.Forms/BuildIt.Forms.Controls/Platforms/Android/CameraCaptureListener.cs

```diff
@@ -0,0 +1,90 @@
+﻿using Android.Hardware.Camera2;
+using Java.Lang;
+
+namespace BuildIt.Forms.Controls.Platforms.Android
+{
+    public class CameraCaptureListener : CameraCaptureSession.CaptureCallback
+    {
+        private readonly CameraPreviewControlRenderer owner;
+
+        public CameraCaptureListener(CameraPreviewControlRenderer owner)
+        {
+            if (owner == null)
+            {
+                throw new System.ArgumentNullException("owner");
+            }
+
+            this.owner = owner;
+        }
+
+        public override void OnCaptureCompleted(CameraCaptureSession session, CaptureRequest request, TotalCaptureResult result)
+        {
+            Process(result);
+        }
+
+        public override void OnCaptureProgressed(CameraCaptureSession session, CaptureRequest request, CaptureResult partialResult)
+        {
+            Process(partialResult);
+        }
+
+        private void Process(CaptureResult result)
+        {
+            switch (owner.State)
+            {
+                case CameraPreviewControlRenderer.STATE_WAITING_LOCK:
+                    {
+                        Integer afState = (Integer)result.Get(CaptureResult.ControlAfState);
+                        if (afState == null)
+                        {
+                            owner.CaptureStillPicture();
+                        }
+                        else if ((afState.IntValue() == ((int)ControlAFState.FocusedLocked)) ||
```

Comment:
Can we case afState to a ControlAFState and tidy up this logic?

Reply:
can take a look but afState is a java integer so i'd imagine doing so would require Java-C# interop. should i go ahead and make that change?

Reply:
actually, just realised afState gets turned to a C# int primitive anyway so will go ahead and make the change